docs/source/primitives.rst

```diff
@@ -286,6 +286,99 @@ Schema
 
 .. jsonschema:: ../../src/core/schema/components/component.json
 
+.. jsonschema:: ../../src/core/schema/common/component_lookup.json
+
+Button
+------
+
+.. jsonschema:: ../../src/core/schema/components/button.json
+
+Triggers action on click. The default buttons are `submit`, `next`, `back`. A next button can set a value in the context, allowing for branching of flows based on button presses.
+
+Actions
+^^^^^^^
+
+* `submit` moves to the next task while saving the contents of the form
+* `next` moves to the next task while ignoring form values
+* `back` moves to the previous screen
+
+.. .. code-block::
+..     {
+..         "type": "button",
+..         "action": "next",
+..         "style": "primary",
+..         "text": "Reset",
+..         "value": false,
+..         "destination_path": "$.save"
+..     }
+
+
+
+Data display
+------------
+
+Display data, this can take in a list of strings or objects containing label and value keys.
+
+.. jsonschema:: ../../src/core/schema/components/data_display.json
+
+
+Input
+-----
+
+Used to collect text/number information from the user. The input component can also create or update
+an object based on its `value`. To build an output json object use `output_ref` and `output` where
+`output` is the structure of the output object with any default values and `output_ref` is the
+`jsonpath` to insert the `value` from the input field. If you wish to select a known object use
+`input_key` and `input_ref`, where `input_key` is a `jsonpath` pointing to a list of objects and
+`input_ref` is the attribute on the object to match the `value` against. There is nothing stopping
+from using both input and output pairs, in this case the field will first try to select using `input_key`
+and `input_ref` then, if nothing is found, will fallback to building an object using `output` and `output_ref`.
+
+.. jsonschema:: ../../src/core/schema/components/input.json
+
+
+Message box
+-----------
+
+A styled text box for presenting information to the user such as warning, errors, or instructions.
+
+.. jsonschema:: ../../src/core/schema/components/message_box.json
+
+
+Toggle
+------
+
+A switch element.
+
+.. jsonschema:: ../../src/core/schema/components/toggle.json
+
+Checkbox
+--------
+
+A selectable list.
+
+.. note:: If two check boxes have the same value selecting on will select the other.
```

Comment:
what does this mean? when you check a box where two act on the same object it will act on both? We can clarify this